adapters/tlsio_openssl.c

```diff
@@ -515,19 +516,26 @@ static void openssl_static_locks_uninstall(void)
 {
     if (openssl_locks != NULL)
     {
-        int i;
-        CRYPTO_set_locking_callback(NULL);
-
-        for (i = 0; i < CRYPTO_num_locks(); i++)
+        if (locks_init_count <= 0)
```

Comment:
><= [](start = 29, length = 2)

I think < 0 is an error actually showing unbalanced install/uninstall calls.
So I think there should be 3 possible cases treated here:
< 0 => error
== 0 => cleanup
> 0 => decrement the nesting counter

Reply:
Fixed.